demisto_sdk/commands/common/tools.py

```diff
@@ -2818,9 +2823,23 @@ def compare_context_path_in_yml_and_readme(yml_dict, readme_content):
     return different_contexts
 
 
-def write_yml(yml_path: str, yml_data: Dict):
-    with open(yml_path, "w") as f:
-        yaml.dump(yml_data, f)  # ruamel preservers multilines
+def write_dict(
+    path: Union[Path, str],
+    data: Dict,
+    handler: Optional[XSOAR_Handler] = None,
+    indent: int = 0,
+    sort_keys: bool = False,
+    **kwargs,
+):
+    """
+    Write unicode content into a json/yml file.
+    """
+    path = Path(path)
+    file_handler = handler or (json if path.suffix.lower() == ".json" else yaml)
```

Comment:
the `else` is risky (a `.md` for instance would be given the yaml loader), i'd replace it with two `if` and raise if it's not a supported suffix